### DISCUSSION TITLE
Add a device class for measurements of Money

### DISCUSSION THREAD

**Discussion by carlos-sarmiento** [home-assistant/architecture#1049] [Entity Models]
I am part of the group of people who want to be able to use long term statistics to keep track of amounts of money. 

The maintainers have been pretty clear that adding the 'measurement' state class to the 'monetary' device class is a non starter, as that device class is used to keep track of accumulated totals. 

So I wanted to explore if there would be an appetite for new device classes to represent situations in which the amount of money is not a total but a fluctuating amount.

In particular I am thinking of two separate but similar situations
- Rates: representing the cost of something. It would take a compound unit (money/quantity).
- Balances: representing the fluctuating balance on an account (bank accounts, debts, stocks, etc.). Would have a simple monetary unit.

The idea of these two is that they better represent entities that already exist on HA (dynamic pricing for energy being a simple example) or things that people are interested in tracking and automating around (warn me when my account balance has gone below a threshold so I stop spending money I shouldn't) 

Would this be an acceptable proposal for the project?

**Comment by gjohansson-ST** [comment]
Balances are the `monetary` class which is already total so you're then proposing something which is already there. For example the money you have on a certain bank account is a total. It's the accumulation of multiple transactions ending up in a certain balance.
Also rates is something different (which you're also writing as the UoM you mention is money/"something".

I suppose what you mean is actually that you want to get the `mean`/`min`/`max` that `measurement` provides?

**Reply by carlos-sarmiento** [reply]
My end goal is to get min/max/average and avoid the semantics of last_reset.
